src/GitVersionCore/BuildServers/GitHubActions.cs

```diff
@@ -0,0 +1,74 @@
+using GitVersion.Logging;
+using GitVersion.OutputVariables;
+using GitVersion.Configuration;
+
+namespace GitVersion.BuildServers
+{
+    public class GitHubActions: BuildServerBase
+    {
+        private readonly IConsole console;
+        // https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables#default-environment-variables
+
+        public GitHubActions(IEnvironment environment, ILog log, IConsole console) : base(environment, log)
+        {
+            this.console = console;
+        }
+
+        public const string EnvironmentVariableName = "GITHUB_ACTION";
+        protected override string EnvironmentVariable { get; } = EnvironmentVariableName;
+        public override bool CanApplyToCurrentContext()
+        {
+            return !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(EnvironmentVariable));
+        }
+
+        public override string GenerateSetVersionMessage(VersionVariables variables)
+        {
+            // There is no equivalent function in GitHub Actions.
+
+            return string.Empty;
+        }
+
+        public override string[] GenerateSetParameterMessage(string name, string value)
+        {
+            // https://help.github.com/en/actions/automating-your-workflow-with-github-actions/development-tools-for-github-actions#set-an-environment-variable-set-env
+            // Example
+            // echo "::set-env name=action_state::yellow"
+
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                var key = $"GitVersion_{name}";
+
+                console.WriteLine($"::set-env name={key}::{value}");
+
+                return new[]
+                {
+                    $"Adding Environment Variable to future steps. name='{key}' value='{value}'"
+                };
+            }
+
+            return new string[0];
```

Comment:
I think we need to use the same approach as for the [AzurePipilines ](https://github.com/GitTools/GitVersion/blob/master/src/GitVersionCore/BuildServers/AzurePipelines.cs#L22-L25)
```suggestion
return new[]
{
    $"::set-env name=GitVersion_{name}::{value}"
};
```

Reply:
No problem, I had it this way first, just wasn't sure what would be right.